src/core/endpoints/subscriptionUtils/receiveMessages.js

```diff
@@ -56,8 +56,9 @@ const endpoint = {
         issuingClientId: envelope.i,
         subscribeKey: envelope.k,
         originationTimetoken: envelope.o,
+        userMetadata: envelope.u,
         publishMetaData: {
-          timetoken: envelope.p.t,
```

Comment:
Is there a reason to rename field? It is alsready inside of `publishMetaData` and it doesn't look like that we need to clarify once again that it is `publishTimetoken`. Is there other reason for it?

Reply:
It was typo before.
In eventEmitter (the same code was there in subscriptionManager before) while creating event information object, [this](https://github.com/pubnub/javascript/blob/26bb23c85943d54094139b4639736b344db26c63/src/core/components/eventEmitter.js#L69) was the way we used to assign.

so instead of chaning property name in all places (which may be error prone like we have now with channel/membership) I rename it here in subscription utility method.. same as older endpoint implementation [here](https://github.com/pubnub/javascript/blob/26bb23c85943d54094139b4639736b344db26c63/src/core/endpoints/subscribe.js#L64) 

in short, It's renaming to make is compatible with old implementation logic

Reply:
this can be refactored other way around than as it's there!

Reply:
@parfeon  refactored!